### PR TITLE
Cleaned up codebase of components

### LINKED ISSUE #906: Clean up codebase of Components
Cleanup code for Components from `Accordion ` to `Input` in the order of listing in Storybook
@vysakh-m _a

### PR DESCRIPTION
**Pull request by vysakh-m**
Fixes #906 

- Cleaned up the first eleven components in the order of listing in Storybook (`Accordion`, `ActionDropdown`, `Avatar`, `Button`, `Callout`, `Checkbox`, `ColorPicker`, `Date & Time Picker`, `Dropdown`, `EmailInput`, `Input`)

### CODE REVIEW

**Comment by vysakh-m** [comment]
@amaldinesh7 _a Please review

File: lib/components/Button.js [hunk 1]
```diff
@@ -59,9 +53,16 @@ const Button = React.forwardRef((props, ref) => {
     };
   }
 
-  let Icon =
+  const handleClick = (e) => !loading && !disabled && onClick(e);
```
Comment (amaldinesh7) [inline lib/components/Button.js:56]: This will result in assigning a boolean value a function if the loading is true, which is not right. Please revert the change.

File: lib/components/ColorPicker/Palette.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import classnames from "classnames";
+import PropTypes from "prop-types";
+
+const Palette = ({ color, colorList, onChange }) => (
```
Comment (amaldinesh7) [inline lib/components/ColorPicker/Palette.js:5]:
```suggestion
const Palette = ({ color, colorList=[], onChange }) => (
```

**Review by amaldinesh7 — APPROVED** [review]
@vysakh-m _a Few minor suggestions, please fix it and proceed with merging.
